com.microsoft.mrtk.graphicstools.unity/Editor/Experimental/AmbientOcclusion/AmbientOcclusionTool.cs

```diff
@@ -0,0 +1,325 @@
+            }
+            newMaterial.EnableKeyword("_DIRECTIONAL_LIGHT");
+            newMaterial.EnableKeyword("_SPHERICAL_HARMONICS");
+            newMaterial.SetFloat("_SphericalHarmonics", 1);
+        }
+
+        private Mesh DeepCopyMesh(Mesh source)
+        {
+            var result = new Mesh();
+            result.name = source.name;
+            result.vertices = source.vertices;
+            result.normals = source.normals;
+            result.uv = source.uv;
+            result.uv2 = source.uv2;
+            result.uv3 = source.uv3;
+            result.uv4 = source.uv4;
+            result.uv5 = source.uv5;
+            result.uv6 = source.uv6;
+            result.uv7 = source.uv7;
+            result.uv8 = source.uv8;
+            result.triangles = source.triangles;
+            result.tangents = source.tangents;
+            return result;
+        }
+
+        internal string ValidateMaterialSetup(GameObject go)
+        {
+            var result = "";
+            bool didConversion = false;
+            // For the results to be visible to the user
+            // we need to be using the graphics tools standard shader
+            // and the appropriate parameter must be enabled
+            if (go.GetComponent<MeshRenderer>() is MeshRenderer meshRenderer)
+            {
+                // No material? No problem.
+                if (meshRenderer.sharedMaterials.Length == 0)
+                {
+                    meshRenderer.sharedMaterial = ConfiguredAoMaterial(meshRenderer.gameObject.name);
+                }
+                foreach (var sharedMat in meshRenderer.sharedMaterials)
+                {
+                    // Got non-standard manterials that don't support AO?
+                    if (!StandardShaderUtility.IsUsingGraphicsToolsStandardShader(sharedMat))
+                    {
+                        Debug.LogWarning($"No Graphics Tools Standard material found with 'Vertex ambient occlusion' enabled on {go.name}.", go);
+                        // This is a special interal material, we need something better.
+                        if (sharedMat.name == "Default-Material")
+                        {
+                            meshRenderer.material = ConfiguredAoMaterial(sharedMat.name);
+                        }
+                        if (settings._upgradeMaterials)
+                        {
+                            Debug.LogWarning($"{sharedMat.name} needs upgrade to Graphics Tools Standard! Please do so manually in the inspector.",
+                                             meshRenderer);
+                            Debug.LogWarning("Tip: Don't forget to enable AO!", meshRenderer);
+                            Debug.LogWarning("Tip: If textures look weird, try setting standad-shader.tiling.y to -1", meshRenderer);
+                            // I want to use this but don't have an instance...
```

Comment:
Would love for this to work without having the user have to do the step manually...

Reply:
Works now